### PR TITLE
fix: Using repo_config parameter in teardown to allow for feature-store-yaml overrides

### PR DESCRIPTION
**Pull request by danbaron63**
# What this PR does / why we need it:

* Allows the use of `--feature-store-yaml` when running `teardown` (`feast --feature-store-yaml <some-other-file> teardown`) as this currently does not work as the `.yaml` file is resolved differently to `plan` and `apply`.
* Fixes some issues with linting and formatting that prevent committing & pushing.

# Which issue(s) this PR fixes:

tbd

# Misc

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,9 +16,6 @@
 [![License](https://img.shields.io/badge/License-Apache%202.0-blue)](https://github.com/feast-dev/feast/blob/master/LICENSE)
 [![GitHub Release](https://img.shields.io/github/v/release/feast-dev/feast.svg?style=flat&sort=semver&color=blue)](https://github.com/feast-dev/feast/releases)
 
-## Join us on Slack!
```
Comment (franciscojavierarceo) [inline README.md:19]: Can you revert this change?

**Review by franciscojavierarceo — CHANGES_REQUESTED** [review]
Thanks for making this! Left a request for some small changes.

File: README.md (hunk 1 above)
Comment (danbaron63) [reply README.md:19]: Ah sorry - don't know where that came from - reverted it!

Comment (danbaron63) [reply README.md]: Okay I was able to remove this but only using `--no-verify`

**Comment by danbaron63** [comment]
> Thanks for making this! Left a request for some small changes.

Thanks for reviewing so quickly, wasn't sure if this was intended behaviour on `teardown` but this does appear to fix the issue.

I'm unable to run all tests locally without at least a couple of failures (I'm on M1 mac which may not help :/) - would appreciate if someone could test this as well as I'm also relatively new to this code base!

Thanks again!

File: sdk/python/tests/integration/feature_repos/universal/data_sources/file.py [hunk 2]
```diff
@@ -382,13 +383,13 @@ def setup(self, registry: RegistryConfig):
...
             yaml.dump(config.model_dump(by_alias=True), outfile)
-        repo_path = str(repo_path.resolve())
+        repo_path = Path(str(repo_path.resolve()))
 
         self.server_port = free_port()
         host = "0.0.0.0"
         cmd = [
             "feast",
-            "-c" + repo_path,
+            f"-c {repo_path}",
```
Comment (franciscojavierarceo) [inline sdk/python/tests/integration/feature_repos/universal/data_sources/file.py:392]: it looks like my suggestion here may have broken things

Comment (franciscojavierarceo) [reply sdk/python/tests/integration/feature_repos/universal/data_sources/file.py:392]: You can revert it, sorry!

**Review by franciscojavierarceo — APPROVED** [review]
lgtm!

**Comment by franciscojavierarceo** [comment]
Looks like this breaks the postgres  `sdk/python/tests/integration/registration/test_universal_registry.py` test. Do you mind digging into this? Running the tests locally should work with the `test-python-integration-local`. You can also just run the following command:
```bash
FEAST_IS_LOCAL_TEST=True \
FEAST_LOCAL_ONLINE_CONTAINER=True \
python -m pytest -n 8 --color=yes --integration --durations=5 --dist loadgroup \
-k "not test_lambda_materialization and not test_snowflake_materialization" \
sdk/python/tests/integration/registration/test_universal_registry.py
```
